plugin/clone/include/clone_common.h

```diff
@@ -25,7 +26,37 @@ class Ha_clone_common_cbk : public Ha_clone_cbk {
  public:
   [[nodiscard]] int precopy(THD *thd, uint task_id) override;
 
-  [[nodiscard]] int synchronize_engines() override;
+ protected:
+  /** Perform the cross-engine synchronization for logs: execute a
+    performance_schema.log_status query, and call set_log_stop for each storage
+    engine with its part of STORAGE_ENGINES column JSON object from that query.
+    Also assign synchronization coordinates get from log_status table to the
+    parameter synchronization_coordinates.
+    @param[in]	synchronization_coordinates	synchronization coordinates
+    @return error code */
+  [[nodiscard]] int synchronize_logs(Key_Values &synchronization_coordinates);
+
+ private:
+  /** Get json object from the json wrapper after doing json validations
+    @param[in]	object_name the object name, used for debug printing
+    @param[in]	json_wrapper json wrapper to be converted to be json object
+    @return converted json object */
+  [[nodiscard]] const Json_object *get_json_object(std::string_view object_name,
```

Comment:
It looks like the result of this method can never be a nullptr - if true, return const reference

Reply:
what's your opinion on returning by value and let RVO take care of the copy? here it is not returning a class variable but json_wrapper.get_dom(); Not sure if returning by reference here is the best practice

Reply:
I like RVO and returning the value, but here it wouldn't apply: `json_wrapper.get_dom()` returns a pointer to its internals. If we dereference it and return by value, we copy, nothing to optimize away.